src/shared/test/bridge-test.js

```diff
@@ -10,246 +10,476 @@ describe('shared/bridge', () => {
+            done();
+          }
+        };
+
+        bridge.onConnect(callback);
+        bridge.onConnect(callback);
+        channel = createChannel(port1);
+        createChannelFromBridge(createBridge(), port2);
+      });
+    });
+
+    describe('#on', () => {
+      it('adds a method to the method registry', () => {
+        const myFunction = () => 'hello world';
+        bridge.on('message1', myFunction);
+        assert.equal(bridge.channelListeners.message1, myFunction);
+      });
+
+      it('only allows registering a method once', () => {
+        bridge.on('message1', sandbox.spy());
+        assert.throws(() => bridge.on('message1', sandbox.spy()));
+      });
+    });
+
+    describe('#off', () =>
+      it('removes the method from the method registry', () => {
+        bridge.on('message1', sandbox.spy());
+        bridge.off('message1');
+        assert.isUndefined(bridge.channelListeners.message1);
+      }));
+
+    describe('#call', () => {
+      it('forwards the call to every created channel (with Promise)', async () => {
+        const messageChannel = new MessageChannel();
+        createChannel(port1);
+        createChannel(messageChannel.port1);
+
+        const reciprocalBridge1 = createBridge();
+        const reciprocalBridge2 = createBridge();
+
+        reciprocalBridge1.on('method1', (arg, cb) => cb(null, `${arg}foo`));
+        reciprocalBridge2.on('method1', (arg, cb) => cb(null, `${arg}bar`));
+
+        createChannelFromBridge(reciprocalBridge1, port2);
+        createChannelFromBridge(reciprocalBridge2, messageChannel.port2);
+
+        const results = await bridge.call('method1', 'params1');
+        assert.deepEqual(results.sort(), ['params1foo', 'params1bar'].sort());
+      });
+
+      it('forwards the call to every created channel (with callback)', done => {
+        const messageChannel = new MessageChannel();
+        createChannel(port1);
+        createChannel(messageChannel.port1);
+
+        const reciprocalBridge1 = createBridge();
+        const reciprocalBridge2 = createBridge();
+
+        reciprocalBridge1.on('method1', (arg, cb) => cb(null, `${arg}foo`));
+        reciprocalBridge2.on('method1', (arg, cb) => cb(null, `${arg}bar`));
+
+        createChannelFromBridge(reciprocalBridge1, port2);
+        createChannelFromBridge(reciprocalBridge2, messageChannel.port2);
+
+        bridge.call('method1', 'params1', (err, results) => {
+          assert.deepEqual(results.sort(), ['params1foo', 'params1bar'].sort());
+          assert.isNull(err);
           done();
+        });
+      });
+
+      it('provides a timeout', done => {
+        createChannel(port1);
+        const clock = sinon.useFakeTimers();
+
+        try {
+          bridge.call('method1', 'params1', (err, results) => {
+            assert.deepEqual(results, [null]);
+            assert.isNull(err);
+            done();
+          });
+          clock.tick(1000);
+        } finally {
+          clock.restore();
```

Comment:
I agree with you, it is surprising. Like you, I asked myself why not to reject the promise instead of returning a `[null]`. Rejecting the promise causes the channel to be closed, which is also surprising.